### PR TITLE
Interior blocks

### PR DESCRIPTION
**Pull request by eirikh1996**
### Describe in detail what your pull request accomplishes
This adds the interiorBlocks functionality to Movecraft. interiorBlocks are like allowedBlocks, but those blocks will only be detectable within a space enclosed to the bottom and the sides.

### Checklist
- [x] Proper internationalization
- [x] Tested

### CODE REVIEW

File: modules/Movecraft/src/main/resources/types/Airship.craft [hunk 1]
```diff
@@ -1,141 +0,0 @@
...
-        - N50
-    "#anvil": # anvils are OP armor, so limit it
-        - 0.0
-        - 1.0
-    obsidian: # so is obsidian
-        - 0.0
-        - 1.0
-    water: # so is water
-        - 0.0
-        - 1.0
```
Comment (TylerS1066) [inline modules/Movecraft/src/main/resources/types/Airship.craft:141]: Same here, please rebase.

File: modules/api/src/main/java/net/countercraft/movecraft/craft/Craft.java [hunk 2]
```diff
@@ -1,266 +1,266 @@
...
-    void setAudience(Audience audience);
-
-    Counter<Material> getMaterials ();
-
-    void updateMaterials (Counter<Material> materials);
-
-    double getTotalFuel ();
-
-    void setTotalFuel (double fuel);
-}
```
Comment (TylerS1066) [inline modules/api/src/main/java/net/countercraft/movecraft/craft/Craft.java:266]: Same here, please rebase.

**Review by TylerS1066 — CHANGES_REQUESTED** [review]

**Review by oh-noey — COMMENTED** [review]
Reviewed with detection specific comments. I haven't looked at anything else.

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/processing/tasks/detection/DetectionTask.java [hunk 3]
```diff
@@ -299,6 +290,50 @@ public Effect get() {
...
+                if (neighbor.subtract(poll).getY() < 0)
+                    continue;
+                if (visited.contains(poll))
+                    continue;
+                exterior.add(neighbor);
+            }
+            visited.add(poll);
+        }
+        final HitBox interior = invertedHitbox.difference(exterior);
+        final InteriorBlockValidator validator = new InteriorBlockValidator();
```
[the remark this answers is not in the document]
Comment (eirikh1996) [reply modules/Movecraft/src/main/java/net/countercraft/movecraft/processing/tasks/detection/DetectionTask.java]: done

**Comment by eirikh1996** [comment]
Requested changes made @TylerS1066

**Review by TylerS1066 — CHANGES_REQUESTED** [review]
I don't know enough about the details of the processing system to do a through review, but with Hazel's comments and knowing the issues with the water detection system, this algorithm needs to be re-thought.

**Pull request closed without merging**